pkg/ssh/connection_golang.go

```diff
@@ -272,23 +273,54 @@ func ValidateAndConfigure(uri *url.URL, iden string) (*ssh.ClientConfig, error)
 	if err != nil {
 		return nil, err
 	}
-	keyFilePath := filepath.Join(homedir.Get(), ".ssh", "known_hosts")
-	known, err := knownhosts.New(keyFilePath)
-	if err != nil {
-		return nil, fmt.Errorf("creating host key callback function for %s: %w", keyFilePath, err)
-	}
 
+	var keyErr *knownhosts.KeyError
 	cfg := &ssh.ClientConfig{
-		User:            uri.User.Username(),
-		Auth:            authMethods,
-		HostKeyCallback: known,
-		Timeout:         tick,
+		User: uri.User.Username(),
+		Auth: authMethods,
+		HostKeyCallback: ssh.HostKeyCallback(func(host string, remote net.Addr, pubKey ssh.PublicKey) error {
+			keyFilePath := filepath.Join(homedir.Get(), ".ssh", "known_hosts")
+			known, err := knownhosts.New(keyFilePath)
+			if err != nil {
+				if os.IsNotExist(err) && insecureIsMachineConnection {
+					logrus.Warn("please create a known_hosts file and recreate the machine to add the public key entry")
```

Comment:
I can see nothing in the existing Podman codebase, this PR, nor https://github.com/containers/podman/pull/15471, that would add a public key entry when creating a machine. What am I missing?

Reply:
Sorry, this log might be misleading. I will repush removing this. It is leftover from the old version thank you for catching this